classes/class-ledyer-main.php

```diff
@@ -153,15 +153,17 @@ public function update_order_status( $request ) {
 					$order = wc_get_order( $order->ID );
 
 					if( 'com.ledyer.order.create' === $request_body['eventType'] ) {
-						// TODO: if order was on hold or pending and notification is received, set to processing (update_status)
-
-						$order->payment_complete( $order_id );
+						$order->update_status('pending');
 						$response = ledyer()->api->acknowledge_order( $order_id );
 						if( is_wp_error( $response ) ) {
 							\Ledyer\Logger::log( 'Couldn\'t acknowledge order ' . $order_id  );
 						}
 					}
 
+					if( 'com.ledyer.order.ready_for_capture' === $request_body['eventType'] ) {
+						$order->update_status('processing');
+					}
+
```

Comment:
Unless we receive the `ready_for_capture` event, the order status should be pending -from the point that we receive the order created event - otherwise put it on hold.

Only update order status here going forward in order to avoid race conditions between several classes wanting to update the order status simultaneously.